internal/services/managedhsm/key_vault_managed_hardware_security_module_key_rotation_policy_resource.go

```diff
@@ -0,0 +1,288 @@
+			}
+
+			id, err := parse.ManagedHSMDataPlaneVersionlessKeyID(metadata.ResourceData.Id(), domainSuffix)
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.GetKeyRotationPolicy(ctx, id.BaseUri(), id.KeyName)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+			schema := flattenKeyRotationPolicy(resp)
+			schema.ManagedHSMKeyID = id.ID()
+
+			return metadata.Encode(&schema)
+		},
+	}
+}
+
+func (r KeyVaultMHSMKeyRotationPolicyResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ManagedHSMs.DataPlaneRoleAssignmentsClient
+			domainSuffix, ok := metadata.Client.Account.Environment.ManagedHSM.DomainSuffix()
+			if !ok {
+				return fmt.Errorf("could not determine Managed HSM domain suffix for environment %q", metadata.Client.Account.Environment.Name)
+			}
+
+			var config MHSMKeyRotationPolicyResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id, err := parse.ManagedHSMDataPlaneVersionlessKeyID(metadata.ResourceData.Id(), domainSuffix)
+			if err != nil {
+				return err
+			}
+
+			if _, err := client.UpdateKeyRotationPolicy(ctx, id.BaseUri(), id.KeyName, expandKeyRotationPolicy(config)); err != nil {
+				return fmt.Errorf("updating HSM Key Rotation Policy for Key %q: %v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r KeyVaultMHSMKeyRotationPolicyResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ManagedHSMs.DataPlaneKeysClient
+
+			domainSuffix, ok := metadata.Client.Account.Environment.ManagedHSM.DomainSuffix()
+			if !ok {
+				return fmt.Errorf("could not determine Managed HSM domain suffix for environment %q", metadata.Client.Account.Environment.Name)
+			}
+
+			id, err := parse.ManagedHSMDataPlaneVersionlessKeyID(metadata.ResourceData.Id(), domainSuffix)
+			if err != nil {
+				return err
+			}
+
+			// settign a blank policy to delete the existing policy
+			if _, err := client.UpdateKeyRotationPolicy(ctx, id.BaseUri(), id.KeyName, keyvault.KeyRotationPolicy{
+				LifetimeActions: pointer.To([]keyvault.LifetimeActions{}),
+				Attributes:      &keyvault.KeyRotationPolicyAttributes{},
+			}); err != nil {
+				return fmt.Errorf("deleting HSM Key Rotation Policy for Key %q: %v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandKeyRotationPolicy(policy MHSMKeyRotationPolicyResourceSchema) keyvault.KeyRotationPolicy {
+
+	var expiryTime *string // = nil // needs to be set to nil if not set
+	if policy.ExipreAfter != "" {
+		expiryTime = utils.String(policy.ExipreAfter)
```

Comment:
Can we not use the `utils` package here as we're deprecating these functions, and instead use the `pointer` package as you have done in the flatten below?
```suggestion
		expiryTime = pointer.To(policy.ExipreAfter)
```